contracts/FIFOMarket.sol

```diff
@@ -147,27 +148,41 @@ contract FIFOMarket is
         suppliers[i] = supplier;
         remainingAmountToFill -= removalAmount;
       }
-
+      numberOfRemovals++;
       if (remainingAmountToFill == 0) {
         break;
       }
     }
 
+    uint256[] memory batchedIds = new uint256[](numberOfRemovals);
+    uint256[] memory batchedAmounts = new uint256[](numberOfRemovals);
+
+    for (
+      uint256 i = _queueHeadIndex;
+      i < _queueHeadIndex + numberOfRemovals;
+      i++
+    ) {
+      batchedIds[i] = ids[i];
+      batchedAmounts[i] = amounts[i];
+    }
+
     bytes memory encodedCertificateAmount = abi.encode(certificateAmount);
-    _certificate.mintBatch(recipient, ids, amounts, encodedCertificateAmount);
-    for (uint256 i = 0; i < ids.length; i++) {
-      if (amounts[i] == 0) {
```

Comment:
This is no longer necessary since we're chopping off all items from the array that would otherwise have a value of 0